### PR TITLE
Adds additional Dapr Endpoint env var support

### LINKED ISSUE #577: Support DAPR_HTTP_ENDPOINT and/or DAPR_GRPC_ENDPOINT
## Describe the proposal

Allow consistent way to configure endpoint for Dapr's sidecar, while enabling remote sidecar configuration.

Child of https://github.com/dapr/dapr/issues/6035

SDK should parse these in a consistent way, so operators have a consistent environment variable to set for Dapr apps. Right now, each SDK has their own way of configuring the endpoint for Dapr's sidecar.

If SDK allows gRPC communication from SDK to sidecar, then `DAPR_GRPC_ENDPOINT` must be supported.
If SDK allows HTTP communication from SDK to sidecar, then `DAPR_HTTP_ENDPOINT` must be supported.
DAPR_HTTP_ENDPOINT and DAPR_GRPC_ENDPOINT and not exclusive, meaning it is acceptable to support both, like in the Java SDK, if the SDK supports both protocols.
Also, companies might decide to offer different endpoints per protocol, so there is not an unique environment variable for both.

### PR DESCRIPTION
**Pull request by berndverst**
# Description

Adds support for `DAPR_GRPC_ENDPOINT` and `DAPR_HTTP_ENDPOINT` in addition to the existing `DAPR_RUNTIME_HOST` environment variables to support talking to remote Dapr instances.

Fixes #577

Also bumps the development version of the SDKs!

### CODE REVIEW

**Comment by berndverst** [comment]
With env var set:

```
>>> os.environ.update({"DAPR_GRPC_ENDPOINT": "http://bernd.dev"})
>>> from dapr.clients import DaprClient
>>> with DaprClient() as c:
...   c.get_metadata()
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "/Users/bverst/src/dapr/python-sdk/dapr/clients/grpc/client.py", line 1441, in get_metadata
    _resp, call = self._stub.GetMetadata.with_call(GrpcEmpty())
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/bverst/.pyenv/versions/3.11.3/lib/python3.11/site-packages/grpc/_channel.py", line 1043, in with_call
    return _end_unary_response_blocking(state, call, True, None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/bverst/.pyenv/versions/3.11.3/lib/python3.11/site-packages/grpc/_channel.py", line 910, in _end_unary_response_blocking
    raise _InactiveRpcError(state)  # pytype: disable=not-instantiable
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
        status = StatusCode.UNAVAILABLE
        details = "DNS resolution failed for http://bernd.dev:50001: C-ares status is not ARES_SUCCESS qtype=AAAA name=http://bernd.dev:50001 is_balancer=0: Domain name not found"
        debug_error_string = "UNKNOWN:DNS resolution failed for http://bernd.dev:50001: C-ares status is not ARES_SUCCESS qtype=AAAA name=http://bernd.dev:50001 is_balancer=0: Domain name not found {created_time:"2023-06-16T15:28:22.110441-07:00", grpc_status:14}"
```

With env var not set:

```
>>> from dapr.clients import DaprClient
>>> with DaprClient() as c:
...   c.get_metadata()
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "/Users/bverst/src/dapr/python-sdk/dapr/clients/grpc/client.py", line 1441, in get_metadata
    _resp, call = self._stub.GetMetadata.with_call(GrpcEmpty())
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/bverst/.pyenv/versions/3.11.3/lib/python3.11/site-packages/grpc/_channel.py", line 1043, in with_call
    return _end_unary_response_blocking(state, call, True, None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/bverst/.pyenv/versions/3.11.3/lib/python3.11/site-packages/grpc/_channel.py", line 910, in _end_unary_response_blocking
    raise _InactiveRpcError(state)  # pytype: disable=not-instantiable
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
        status = StatusCode.UNAVAILABLE
        details = "failed to connect to all addresses; last error: UNKNOWN: ipv4:127.0.0.1:50001: Failed to connect to remote host: Connection refused"
        debug_error_string = "UNKNOWN:failed to connect to all addresses; last error: UNKNOWN: ipv4:127.0.0.1:50001: Failed to connect to remote host: Connection refused {grpc_status:14, created_time:"2023-06-16T15:30:53.68215-07:00"}"
```

Ignore the errors here - I was not running a sidecar.

File: dapr/aio/clients/grpc/client.py [hunk 1]
```diff
@@ -124,7 +124,8 @@ def __init__(
         """
         useragent = f'dapr-sdk-python/{__version__}'
         if not address:
-            address = f"{settings.DAPR_RUNTIME_HOST}:{settings.DAPR_GRPC_PORT}"
+            address = f'''{settings.DAPR_GRPC_ENDPOINT if settings.DAPR_GRPC_ENDPOINT
```
Comment (artursouza) [inline dapr/aio/clients/grpc/client.py:127]: The environment variable affects port too. See logic done in Java SDK - see how it sets port to 80 or 443 automatically based on the URL schema. It needs URL parsing:

https://github.com/dapr/java-sdk/blob/76aec01e9aa4af7a72b910d77685ddd3f0bf86f3/sdk/src/main/java/io/dapr/client/DaprClientBuilder.java#L177C11-L177C11

File: dapr/clients/http/client.py [hunk 2]
```diff
@@ -50,7 +50,8 @@ def __init__(self,
 
     def get_api_url(self) -> str:
         return 'http://{}:{}/{}'.format(
```
Comment (artursouza) [inline dapr/clients/http/client.py:52]: The new env variable replaces this entire string, not just the host part.

**Review by artursouza — CHANGES_REQUESTED** [review]

**Comment by berndverst** [comment]
Closing this PR because the proposal this is based on was modified.

**Pull request closed without merging**